### PR TITLE
Feat: OPX3.1-dev1

### PR DESCRIPTION
**Pull request by GarrickHe**
* Bugfix: Removed assertion of delay entry in sdi cfg file
* Bugfix: Different optics type detected
* Bugfix: Fix endianess issue
* Update: Set default RX value to conform with dBm measurement

Signed-off-by: Garrick He <garrick_he@dell.com>

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,8 +1,17 @@
+opx-sdi-sys (4.24.0) unstable; urgency=medium
+
+  * Bugfix: Removed assertion of delay entry in sdi cfg file
+  * Bugfix: Different optics type detected
```
Comment (jeffyin-dell) [inline debian/changelog]: Would be good to elaborate on this, as "different optics type detected" doesn't really sound like a bug.

File: inc/Makefile.am [hunk 2]
```diff
@@ -93,6 +93,7 @@ nobase_include_HEADERS = \
         opx/sdi_comm_dev.h \
         opx/sdi_entity.h \
         opx/sdi_entity_info.h \
+	opx/sdi_ext_ctrl.h \
```
Comment (jeffyin-dell) [inline inc/Makefile.am]: the spacing here looks off by 2 spaces

File: src/drivers/sdi_onie_eeprom.c [hunk 3]
```diff
@@ -226,13 +226,13 @@ static void sdi_onie_fill_vendor_extn(sdi_onie_tlv_field *tlv_fld,
                 entity_data->air_flow            = SDI_PWR_AIR_FLOW_REVERSE;
                 break;
             default:
-                err_mesg = "Invalid value for PSU type in ONIE EEPROM vendor extension";
+                err_mesg = "Invlaid value for PSU type in ONIE EEPROM vendor extension";
```
Comment (jeffyin-dell) [inline src/drivers/sdi_onie_eeprom.c]: we're _introducing_ a typo here

File: src/hwcore/unittest_utils/pysdi.py [hunk 4]
```diff
@@ -6,9 +6,9 @@
 # not use this file except in compliance with the License. You may obtain
 # a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 #
-# THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+# THIS CODE IS PROVIDED ON AN #AS IS* BASIS, WITHOUT WARRANTIES OR
```
Comment (jeffyin-dell) [inline src/hwcore/unittest_utils/pysdi.py]: The changes in this file are incorrect and introduce typo/spacing errors.

File: src/hwcore/unittest_utils/sdi_shell.py [hunk 5]
```diff
@@ -6,9 +6,9 @@
 # not use this file except in compliance with the License. You may obtain
 # a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 #
-# THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+# THIS CODE IS PROVIDED ON AN #AS IS* BASIS, WITHOUT WARRANTIES OR
```
Comment (jeffyin-dell) [inline src/hwcore/unittest_utils/sdi_shell.py]: The changes in this file are incorrect and introduce typo/spacing errors.

**Review by jeffyin-dell — CHANGES_REQUESTED** [review]

**Comment by GarrickHe** [comment]
Pull-request updated, HEAD is now 4d9280addd45b149245eb58374fe2719368ff365

**Review by hpersh — DISMISSED** [review]

**Comment by GarrickHe** [comment]
Pull-request updated, HEAD is now 085d17223cf7269b72e376dd1f24c80ac97d1481

**Review by hpersh — APPROVED** [review]

**Review by jeffyin-dell — APPROVED** [review]

**Pull request merged**